### PR TITLE
Fix NamespaceSelectFilter

### PR DESCRIPTION
**Pull request by Nokel81**
- Not displaying checks when "All Namespaces" was clicked

- Not toggling namespaces that are implicitly selected

Signed-off-by: Sebastian Malton <sebastian@malton.name>

## Before

https://user-images.githubusercontent.com/8225332/139443082-4c4916e2-f454-45a3-8ad8-bc7a7dc5629d.mov

## After

https://user-images.githubusercontent.com/8225332/139441866-cfb9f437-06ea-454e-aeb4-067d9724dff9.mov

### CODE REVIEW

**Review by jim-docker — COMMENTED** [review]
works as described, it's very nice to now be able to select all but one by selecting "All namespaces" and then toggling one off.

I find it weird that the dropdown list is reordered while I'm holding down the cmd key. I might like to toggle one off and then back on to see how the list of pods (say) changes but that's a bit cumbersome if it jumps to the bottom when I deselect it. Is there a reason/use case for doing sorting while the cmd key is held? (I think the sorting is ok when the dropdown first appears)

File: src/renderer/components/+namespaces/namespace-select-filter.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ const Placeholder = observer((props: PlaceholderProps<any, boolean>) => {
   const getPlaceholder = (): React.ReactNode => {
     const namespaces = namespaceStore.contextNamespaces;
 
-    if (!namespaceStore.selectedNamespaces.length || !namespaces.length) {
+    if (namespaceStore.areAllSelectedImplicitly || !namespaces.length) {
```
Comment (jim-docker) [inline src/renderer/components/+namespaces/namespace-select-filter.tsx:40]: that's good, makes it clearer what we're testing for

**Comment by Nokel81** [comment]
Good point about the sorting, I don't think there is a good reason for that. Will fix.

**Comment by jim-docker** [comment]
> Good point about the sorting, I don't think there is a good reason for that. Will fix.

It's actually also cumbersome to reselect a bunch from the bottom, the list jumps up every time

**Comment by Nokel81** [comment]
Done, PTALA @jim-docker

**Review by jim-docker — APPROVED** [review]

**Pull request merged**